src/Universum/Monad/Container.hs

```diff
@@ -0,0 +1,150 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Universum.Monad.Container
+       ( concatMapM
+       , concatForM
+
+       , allM
+       , anyM
+       , andM
+       , orM
+
+       , (<$!>)
+       ) where
+
+import Control.Applicative (Applicative (pure))
+import Data.Function ((.))
+import Data.Traversable (Traversable (traverse))
+import Prelude (Bool (..), Monoid, flip)
+
+import Universum.Base (IO, seq)
+import Universum.Container (Container, Element, fold, toList)
+import Universum.Functor (fmap)
+import Universum.Monad.Reexport (Monad (..))
+
+-- | Lifting bind into a monad. Generalized version of @concatMap@
+-- that works with a monadic predicate. Old and simpler specialized to list
+-- version had next type:
+--
+-- @
+--     concatMapM :: Monad m => (a -> m [b]) -> [a] -> m [b]
+-- @
+--
+-- Side note: previously it had type
+--
+-- @
+--     concatMapM :: (Applicative q, Monad m, Traversable m)
+--                => (a -> q (m b)) -> m a -> q (m b)
+-- @
+--
+-- Such signature didn't allow to use this function when traversed container
+-- type and type of returned by function-argument differed.
+-- Now you can use it like e.g.
+--
+-- @
+--     concatMapM readFile files >>= putStrLn
+-- @
+concatMapM
+    :: ( Applicative f
+       , Monoid m
+       , Container (l m)
+       , Traversable l
+       )
+    => (a -> f m) -> l a -> f m
+concatMapM f = fmap fold . traverse f
+{-# INLINE concatMapM #-}
+
+-- | Like 'concatMapM', but has its arguments flipped, so can be used
+-- instead of the common @fmap concat $ forM@ pattern.
+concatForM
+    :: ( Applicative f
+       , Monoid m
+       , Container (l m)
+       , Traversable l
+       )
+    => l a -> (a -> f m) -> f m
+concatForM = flip concatMapM
+{-# INLINE concatForM #-}
+
+-- | Stricter version of 'Data.Functor.<$>'.
+(<$!>) :: Monad m => (a -> b) -> m a -> m b
```

Comment:
I just discovered that this operator is already in `Control.Monad` module. So you can reexport it from there:

https://www.stackage.org/haddock/lts-9.18/base-4.9.1.0/Control-Monad.html#v:-60--36--33--62-

You don't even need `if`s because it appeared in older version of GHC according to this wiki page:

https://wiki.haskell.org/Base_package#Versions